### PR TITLE
Adds a blog post on yum repo layout changes

### PR DESCRIPTION
**Pull request by bmbouter**

### CODE REVIEW

**Comment by bmbouter** [comment]
@tomjohnson1492 I just realized that our mention-bot was spamming you. That [shouldn't happen again](https://github.com/pulp/pulpproject.org/pull/25). Thank you for the wonderful theme.

File: _posts/2016-11-12-yum-repo-layout-changes.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+3. Metadata packages in `repodata` will use the new layout urls allowing Yum and similar clients to
+   continue discovering packages.
+
+Here is the same example from above organized using the new layout:
+
+```
+[vagrant@dev zoo]$ tree
+.
+├── os
```
Comment (mhrivnak) [inline _posts/2016-11-12-yum-repo-layout-changes.md]: The "os" directory is not something pulp will create. See this comment: https://pulp.plan.io/issues/1976#note-42

I see that comment 32 on that issue mentions a directory named "os", but that just happens to be the last directory in the relative path at which the repo is being published. `/path/to/os/` is the root directory for the repo. Pulp only controls what happens within that directory.

Previously there would be files such as:

```
/path/to/os/foo.rpm
/path/to/os/bar.rpm
/path/to/os/repodata/repomd.xml
```

In the future, there will be:

```
/path/to/os/Packages/foo.rpm
/path/to/os/Packages/bar.rpm
/path/to/os/repodata/repomd.xml
```

File: _posts/2016-11-12-yum-repo-layout-changes.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    └── repomd.xml
+```
+
+
+### Motivation
+
+* Having all of the packages in one folder limits the ability to publish or copy Pulp repositories 
+  on some filesystems.
+
+* The `os` and `Packages` folders are a common convention that large rpm repositories commonly use.
```
Comment (mhrivnak) [inline _posts/2016-11-12-yum-repo-layout-changes.md]: Here also, "os" should not be mentioned.

File: _posts/2016-11-12-yum-repo-layout-changes.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+├── walrus-0.71-1.noarch.rpm
+├── walrus-5.21-1.noarch.rpm
+├── whale-0.2-1.noarch.rpm
+├── wolf-9.4-2.noarch.rpm
+└── zebra-0.1-2.noarch.rpm
+```
+
+The 2.12+ published rpm repository will be different in a few ways:
+
+1. Packages will be published under `os/Packages` and will no longer be published in the root of the
```
Comment (mhrivnak) [inline _posts/2016-11-12-yum-repo-layout-changes.md]: See below. "os" should not be included here.

**Review by mhrivnak — CHANGES_REQUESTED** [review]

File: _posts/2016-11-14-yum-repo-layout-changes.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+├── walrus-0.71-1.noarch.rpm
+├── walrus-5.21-1.noarch.rpm
+├── whale-0.2-1.noarch.rpm
+├── wolf-9.4-2.noarch.rpm
+└── zebra-0.1-2.noarch.rpm
+```
+
+The 2.12+ published rpm repository will be different in a few ways:
+
+1. Packages will be published in the `Packages` and will no longer be published in the root of the
```
Comment (mhrivnak) [inline _posts/2016-11-14-yum-repo-layout-changes.md]: s/`Packages`/`Packages` directory/

File: _posts/2016-11-14-yum-repo-layout-changes.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+   repository.
+
+2. Packages will be placed in a folder by the first character of the package name.
+
+3. Metadata packages in `repodata` will use the new layout urls allowing Yum and similar clients to
```
Comment (mhrivnak) [inline _posts/2016-11-14-yum-repo-layout-changes.md]: This could be ambiguous. I'm not sure if by "metadata packages" you mean the XML files, or the individual blocks within the XML that correspond to a package. I would rephrase as "Metadata files in the `repodata` directory will reference the new relative paths..."

File: _posts/2016-11-14-yum-repo-layout-changes.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    └── repomd.xml
+```
+
+
+### Motivation
+
+* Having all of the packages in one folder limits the ability to publish or copy Pulp repositories 
+  on some filesystems.
+
+* The `Packages` folder is a common convention and keeps the repository organized.
```
Comment (tehsmyers) [inline _posts/2016-11-14-yum-repo-layout-changes.md]: Examples might be useful here, such as http://download.fedoraproject.org/pub/fedora/linux//releases/24/Server/x86_64/os/ and http://mirror.centos.org/centos/7/os/x86_64/

**Review by tehsmyers — APPROVED** [review]

**Review by mhrivnak — APPROVED** [review]
